### PR TITLE
Jon/aave/disable-details-buttons

### PR DESCRIPTION
**Pull request by Jon-edge**
- Disable loan action buttons if an ActionProgram is running
- Disable withdraw/repay if there is nothing to withdraw/repay

Simulator Screen Shot - iPhone 8 Plus - 2022-10-14 at 18 51 42

#### PR Dependencies

 none

#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [x] Tested on iOS device
- [ ] Tested on Android device
- [ ] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202949203074528

### CODE REVIEW

File: src/components/scenes/Loans/ManageCollateralScene.tsx
Comment (samholmes) [inline src/components/scenes/Loans/ManageCollateralScene.tsx:238]: We should store the entire `NetworkFeeMap` as state and then derive the `feeNativeAmount` from that state. This way when we go to fix our fee calculation to include all network fee amounts, we would only need to remove the derived value and directly pass the `NetworkFeeMap` object to the component/function handling the calculation of all fees.

File: src/components/scenes/Loans/ManageCollateralScene.tsx
Comment (samholmes) [inline src/components/scenes/Loans/ManageCollateralScene.tsx]: We should include an insufficient funds case to the `sliderDisabled` condition in a later commit.

File: src/components/scenes/Loans/ManageCollateralScene.tsx
Comment (samholmes) [inline src/components/scenes/Loans/ManageCollateralScene.tsx]: Bravo! I love the reduction in complexity made to this component. It's a very useful abstraction now.

File: src/components/scenes/Loans/ManageCollateralScene.tsx
Comment (samholmes) [inline src/components/scenes/Loans/ManageCollateralScene.tsx:98]: Name suggestion: `isAprChangeShown`. I think if the verb is always going to be "is", thne we have to figure out a noun form for any verbs like "show".

File: src/components/scenes/Loans/ManageCollateralScene.tsx
Comment (samholmes) [inline src/components/scenes/Loans/ManageCollateralScene.tsx:67]: We can add `as const` here to make certain the type is strictly this constant value.

File: src/components/scenes/Loans/ManageCollateralScene.tsx
Comment (samholmes) [inline src/components/scenes/Loans/ManageCollateralScene.tsx:65]: This re-ordering change should be squashed into the previous commit.

**Review by samholmes — CHANGES_REQUESTED** [review]
This review is really for https://github.com/EdgeApp/edge-react-gui/pull/3725. I didn't notice the future commit. We should make the PR Dependencies section of the pull-request template more noticeable both for the requester and reviewers.

File: src/components/cards/TappableCard.tsx
Comment (samholmes) [inline src/components/cards/TappableCard.tsx:27]: This should be a function as all `render*` functions are.

File: src/components/scenes/Loans/LoanDetailsScene.tsx
Comment (samholmes) [inline src/components/scenes/Loans/LoanDetailsScene.tsx:192]: Maybe a separate `title` string for this instead of re-using `wc_smartcontract_warning_title`? It could be set to "Waiting on Status" instead of "Warning". What do you think?

File: src/components/scenes/Loans/ManageCollateralScene.tsx
Comment (Jon-edge) [reply src/components/scenes/Loans/ManageCollateralScene.tsx]: Agreed, actually forgot to make a TODO on that to expand the UI design for that case.

In any case, it will be handled here: https://app.asana.com/0/1200382638405084/1203187329419620/f

**Review by samholmes — APPROVED** [review]
Approved _this_ PR with this second review. Though, this is blocked by the upstream PR and that has some feedback. The following comments are optional feedback changes.

File: src/components/scenes/Loans/ManageCollateralScene.tsx
Comment (Jon-edge) [reply src/components/scenes/Loans/ManageCollateralScene.tsx:98]: Yeah, your suggestion is a nicer **sounding** name, but I would like to caution trying to mold this concept into 'proper English.'

'Proper English' is rife with pitfalls in that the English language itself is just plain weird and has a lot of exceptions and arbitrary rules for tense, plurality, etc. French is similar in this regard. As such, IMO:

**Consistency and hierarchy > Nice sounding English**

Less overhead in naming things too if there's a simple system in place without having to wordsmith.

I know also you are a fan of imperatives, so, `isShowApr` might be the best here in that sense

File: src/components/cards/TappableCard.tsx
Comment (Jon-edge) [reply src/components/cards/TappableCard.tsx:27]: Doesn't work here, TS complains about not allowing functions as children

File: src/components/scenes/Loans/LoanDetailsScene.tsx
Comment (Jon-edge) [reply src/components/scenes/Loans/LoanDetailsScene.tsx:192]: Good catch, I threw this in just to satisfy the props generically. "Please wait" perhaps?

File: src/util/ActionProgramUtils.ts
Comment (Jon-edge) [reply src/util/ActionProgramUtils.ts:92]: It's not just wrapping. We need to cast the type because `actions` does not exist on all `ActionOp` types

Comment (Jon-edge) [reply src/util/ActionProgramUtils.ts:92]: This isn't the first time the typing of these guys caused some snags.... I think we need to separate `SeqActionOp` and `ParActionOp` out from the rest. They're not really operations in themselves but organizational nodes. Redefine the ActionOps as `ActionNodes` and `ActionLeafs`?

Just a thought, maybe you have a better idea.

File: src/components/scenes/Loans/ManageCollateralScene.tsx
Comment (Jon-edge) [reply src/components/scenes/Loans/ManageCollateralScene.tsx:233]: I like constants better if it's plain static data

File: src/components/scenes/Loans/ManageCollateralScene.tsx
Comment (Jon-edge) [reply src/components/scenes/Loans/ManageCollateralScene.tsx:65]: Force pushed these changes

File: src/components/scenes/Loans/ManageCollateralScene.tsx
Comment (Jon-edge) [reply src/components/scenes/Loans/ManageCollateralScene.tsx:67]: Force pushed these changes

File: src/components/scenes/Loans/ManageCollateralScene.tsx
Comment (samholmes) [reply src/components/scenes/Loans/ManageCollateralScene.tsx:98]: It's all arguable I suppose. With that said, the claim `Consistency and hierarchy > Nice sounding English` is conjecture without some predicate to back it the inequality. If you don't want to change it, that's A-OK.

**Pull request merged**
